framework/src/base/NonlinearEigenSystem.C

```diff
@@ -119,6 +122,12 @@ NonlinearEigenSystem::solve()
   // Initialize the solution vector using a predictor and known values from nodal bcs
   setInitialSolution();
 
+// In DEBUG mode, Libmesh will check the residual automatically. This may cause
+// an error because B does not need to assembly by default.
+#ifdef DEBUG
```

Comment:
When you say debug mode, do you mean ONLY debug mode, or do you mean "not optimized" mode (i.e. also included DEVEL mode). Can you point me to the libMesh code in question? I can help answer this.

Reply:
I think I meant ONLY debug mode.

https://github.com/libMesh/libmesh/blob/7f3ced2710c67ef3f7d7ba512090259b09997c74/src/solvers/slepc_eigen_solver.C#L510

Reply:
Perfect! Thanks